uniauth/views.py

```diff
@@ -240,7 +273,17 @@ def logout(request):
     setting is true.
     """
     next_page = request.GET.get('next')
-    auth_method = request.session.get('auth-method')
+    api_mode = get_setting('UNIAUTH_API_MODE')
+
+    if api_mode:
+        # get JWT token from authorization headers; .split(' ') to get the token without Bearer prefix
+        base64_token = request.META['HTTP_AUTHORIZATION'].split(' ')[1]
+        # convert the base64 string token into a Refresh Token object
+        token = RefreshToken(base64_token)
+        # get the auth_method from the JWT token
+        auth_method = token['auth_method']
```

Comment:
Some of these comments are pretty obvious. Admittedly, I'm somewhat guilty of this too 😜